src/polyhedron.jl

```diff
@@ -0,0 +1,189 @@
+export ConvexHullLib
+
+type ConvexHullLib <: PolyhedraLibrary
+    precision::Symbol
+
+    function ConvexHullLib(precision::Symbol=:float)
+        if !(precision in [:float, :exact])
+            error("Invalid precision, it should be :float or :exact")
+        end
+        new(precision)
+    end
+end
+
+type ConvexHullPolyhedron{N, T} <: Polyhedron{N, T}
+    ine::Nullable{HRepresentation{N, T}}
+    inel::Nullable{LiftedHRepresentation{N, T}}
+    ext::Nullable{VRepresentation{N, T}}
+    extl::Nullable{LiftedVRepresentation{N, T}}
+    hlinearitydetected::Bool
+    vlinearitydetected::Bool
+    noredundantinequality::Bool
+    noredundantgenerator::Bool
+
+    function ConvexHullPolyhedron(ine::HRepresentation{N, T}, ext::VRepresentation{N, T}, hld::Bool, vld::Bool, nri::Bool, nrg::Bool)
+        new(ine, nothing, ext, nothing, hld, vld, nri, nrg)
+    end
+    function ConvexHullPolyhedron(ine::HRepresentation{N, T})
+        new(ine, nothing, nothing, nothing, false, false, false, false)
+    end
+    function ConvexHullPolyhedron(ext::VRepresentation{N, T})
+        new(nothing, nothing, ext, nothing, false, false, false, false)
+    end
+end
+
+# ine may decompose fast but if ine is nothing I do not want to ask to compute it to see the type it is
+# saying false normally do not give troubles
+decomposedhfast{N, T}(::Type{ConvexHullPolyhedron{N, T}}) = false
+decomposedvfast{N, T}(::Type{ConvexHullPolyhedron{N, T}}) = false
+decomposedhfast{N, T}(p::ConvexHullPolyhedron{N, T}) = decomposedhfast(ConvexHullPolyhedron{N, T})
+decomposedvfast{N, T}(p::ConvexHullPolyhedron{N, T}) = decomposedvfast(ConvexHullPolyhedron{N, T})
+
+# Helpers
+function getine(p::ConvexHullPolyhedron)
+    if isnull(p.ine)
+        if !isnull(p.inel)
+            p.ine = p.inel
```

Comment:
This is the advantage of the `HRepresentation` abstract type. It avoids conversion between internal representation of the polyhedron as much as possible. If two libraries use the same internal representation they can even exchange representation without needing to do conversion.
The `MathProgBase` representation is actually yet another H-representation of a polyhedron and there is the `LPHRepresentation` for it (see [here](https://github.com/blegat/Polyhedra.jl/blob/master/src/lphrep.jl)).
Each H-representation only defines iterators for equalities, inequalities and how to construct them from those iterators and all the conversion between all the H-representation are defined on top of that.
This enables for examples to use any H-representation as an input for e.g. Gurobi. it just converts it to `LPHRepresentation` and then use the `loadproblem!` version of MathProgBase (see [here](https://github.com/blegat/Polyhedra.jl/blob/master/src/lpqp_to_polyhedra.jl)).
This also makes it easy to convert a `JuMP` model to a H-representation without any conversion. We just extract the MathProgBase representation into a `LPHRepresentation` (see [here](https://github.com/blegat/Polyhedra.jl/blob/master/src/jump.jl)).
Currently, the H-representation are `SimpleHRepresentation`, `LiftedHRepresentation`, `LPHRepresentation`, `CDDInequalityMatrix` (internal rep of CDD) and `LRSInequalityMatrix` (internal rep of LRS).